app/services/ads/search-page-ads-context.js

```diff
@@ -1,36 +1,22 @@
 import { Promise } from 'rsvp';
 import Service, { inject as service } from '@ember/service';
+import fetch from 'fetch';
 import Ads from '../../modules/ads';
 
 export default Service.extend({
-  wikiVariables: service(),
   currentUser: service(),
-  wikiUrls: service(),
   fetchService: service('fetch'),
-  adsContextPromise: null,
-
-  init() {
-    this._super(...arguments);
-
-    this.setAdsContextPromise();
-  },
-
-  /**
-   * @returns {Promise}
-   */
-  getAdsContextPromise() {
-    return this.adsContextPromise;
-  },
+  wikiUrls: service(),
+  wikiVariables: service(),
 
   /**
    * @private
    */
-  setAdsContextPromise() {
```

Comment:
Why not set it in `init`? This endpoint will always return the same context for a given wiki.